tests/ui/coercion/coerce-block-tail-83783.stderr

```diff
@@ -6,6 +6,10 @@ LL |     _consume_reference::<i32>(&async { Box::new(7_i32) }.await);
    |
    = note: expected type `i32`
             found struct `Box<i32>`
+help: consider unboxing the value
+   |
+LL |     _consume_reference::<i32>(&*async { Box::new(7_i32) }.await);
```

Comment:
This is not right?

Reply:
Oh, I didn't notice the `.await`. I guess this is valid, but it would be nice to understand *why* this is being suggested now 🤔